File: tiny-event-sourcing-spring-app/src/main/kotlin/ru/quipy/shop/delivery/events/DeliveryChangeStatusEvent.java
package ru.quipy.shop.delivery.events;

import lombok.Getter;
import ru.quipy.core.annotations.DomainEvent;
import ru.quipy.domain.Event;
import ru.quipy.shop.delivery.DeliveryAggregate;
import ru.quipy.shop.delivery.DeliveryStatus;
import sun.awt.util.IdentityLinkedList;

import java.util.UUID;

@Getter
@DomainEvent(name = "DELIVERY_CHANGE_STATUS_EVENT")
public class DeliveryChangeStatusEvent extends Event<DeliveryAggregate> {
    private UUID deliveryId;
    private DeliveryStatus status;

    public DeliveryChangeStatusEvent(UUID id, DeliveryStatus status) {
        super();
        this.deliveryId = id;
        this.status = status;
    }
}

File: tiny-event-sourcing-spring-app/src/main/kotlin/ru/quipy/shop/delivery/events/DeliveryChangeStatusEvent.kt
package ru.quipy.shop.delivery.events

import lombok.Getter
import ru.quipy.core.annotations.DomainEvent
import ru.quipy.domain.Event
import ru.quipy.shop.delivery.DeliveryAggregate
import ru.quipy.shop.delivery.DeliveryStatus

import java.util.UUID

@Getter
@DomainEvent(name = "DELIVERY_CHANGE_STATUS_EVENT")
class DeliveryChangeStatusEvent(
    val deliveryId: UUID,
    var status: DeliveryStatus
) : Event<DeliveryAggregate>(
    id = deliveryId,
    name = "DELIVERY_CHANGE_STATUS_EVENT")
